--- a/m4/gpg-error.m4	2023-04-01 16:19:09.708126000 +1030
+++ b/m4/gpg-error.m4	2023-04-01 16:53:08.453208587 +1030
@@ -60,7 +60,8 @@
 dnl
 dnl _AM_PATH_GPGRT_CONFIG
 AC_DEFUN([_AM_PATH_GPGRT_CONFIG],[dnl
-  AC_PATH_PROG(GPGRT_CONFIG, gpgrt-config, no, [$prefix/bin:$PATH])
+  #AC_PATH_PROG(GPGRT_CONFIG, gpgrt-config, no, [$prefix/bin:$PATH])
+  AC_PATH_PROG(GPGRT_CONFIG, gpgrt-config, no)
   if test "$GPGRT_CONFIG" != "no"; then
     # Determine gpgrt_libdir
     #
@@ -75,8 +76,9 @@
     #   Fedora/openSUSE style: /usr/lib, /usr/lib32 or /usr/lib64
     # It is assumed that CC is specified to the one of host on cross build.
     if libdir_candidates=$(${CC:-cc} -print-search-dirs | \
-          sed -n -e "/^libraries/{s/libraries: =//;s/:/\\
-/g;p;}"); then
+#          sed -n -e "/^libraries/{s/libraries: =//;s/:/\\
+#/g;p;}"); then
+          sed -n -e "/^libraries/{s/libraries: =//;s/:/\n/gp}"); then
       # From the output of -print-search-dirs, select valid pkgconfig dirs.
       libdir_candidates=$(for dir in $libdir_candidates; do
         if p=$(cd $dir 2>/dev/null && pwd); then
@@ -222,4 +224,4 @@
   AC_SUBST(GPG_ERROR_LIBS)
   AC_SUBST(GPG_ERROR_MT_CFLAGS)
   AC_SUBST(GPG_ERROR_MT_LIBS)
-])
\ No newline at end of file
+])