--- a/CMakeLists.txt	2020-10-22 19:36:16.116017000 -0700
+++ b/CMakeLists.txt	2020-10-22 19:47:22.102464809 -0700
@@ -103,10 +103,14 @@
         string(REPLACE "/MDd" "/MTd" CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
         string(REPLACE "/MDd" "/MTd" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
     endif()
 elseif(MINGW) # 2020.10.23
     target_link_libraries (OpenCL cfgmgr32) # 2020.10.23
+    option (OPENCL_ICD_LOADER_DISABLE_OPENCLON12 "Disable support for OpenCLOn12. Support for OpenCLOn12 should only be disabled when building an import lib to link with, and must be enabled when building an ICD loader for distribution!" OFF) # 2020.10.23
+    if (OPENCL_ICD_LOADER_DISABLE_OPENCLON12)  # 2020.10.23
+        target_compile_definitions(OpenCL PRIVATE OPENCL_ICD_LOADER_DISABLE_OPENCLON12)  # 2020.10.23
+    endif()  # 2020.10.23
 else()
     if (APPLE)
         target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
     else ()
         set_target_properties (OpenCL PROPERTIES LINK_FLAGS "-Wl,--version-script -Wl,${CMAKE_CURRENT_SOURCE_DIR}/loader/linux/icd_exports.map")
